plugin/src/main/java/org/opensearch/ml/action/update/UpdateModelTransportAction.java

```diff
@@ -0,0 +1,564 @@
+        try {
+            updateRequest.doc(updateModelInput.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS));
+            updateRequest.docAsUpsert(true);
+            if (isUpdateModelCache) {
+                String[] targetNodeIds = getAllNodes();
+                MLUpdateModelCacheNodesRequest mlUpdateModelCacheNodesRequest = new MLUpdateModelCacheNodesRequest(
+                    targetNodeIds,
+                    modelId,
+                    isPredictorUpdate
+                );
+                client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                    client
+                        .update(
+                            updateRequest,
+                            getUpdateResponseListenerWithUpdateModelCache(modelId, wrappedListener, mlUpdateModelCacheNodesRequest)
+                        );
+                }, e -> {
+                    log
+                        .error(
+                            "Failed to register ML model with model ID {} to the new model group with model group ID {}",
+                            modelId,
+                            newModelGroupId,
+                            e
+                        );
+                    wrappedListener.onFailure(e);
+                }));
+            } else {
+                client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                    client.update(updateRequest, getUpdateResponseListener(modelId, wrappedListener));
+                }, e -> {
+                    log
+                        .error(
+                            "Failed to register ML model with model ID {} to the new model group with model group ID {}",
+                            modelId,
+                            newModelGroupId,
+                            e
+                        );
+                    wrappedListener.onFailure(e);
+                }));
+            }
+        } catch (IOException e) {
+            log.error("Failed to build update request.");
+            wrappedListener.onFailure(e);
+        }
+    }
+
+    private ActionListener<UpdateResponse> getUpdateResponseListenerWithUpdateModelCache(
+        String modelId,
+        ActionListener<UpdateResponse> wrappedListener,
+        MLUpdateModelCacheNodesRequest mlUpdateModelCacheNodesRequest
+    ) {
+        return ActionListener.wrap(updateResponse -> {
+            // The update response returned an unexpected status may indicate a failed update
+            if (updateResponse != null && updateResponse.getResult() != DocWriteResponse.Result.UPDATED) {
+                log.error("Model id:{} failed update", modelId);
+                wrappedListener.onResponse(updateResponse);
+                return;
+            }
+            client.execute(MLUpdateModelCacheAction.INSTANCE, mlUpdateModelCacheNodesRequest, ActionListener.wrap(r -> {
+                log.info("Successfully perform in-place update ML model with model ID {}", modelId);
```

Comment:
Check if all nodes  updated cache successfully, if not, we should return failure `wrappedListener.onFailure( new XXXException("Index updated successfully ,but failed to update model cache on nodes: [nodeid1, nodeid2]}"));`